wasmcloud-wasi-kv/README.md

```diff
@@ -0,0 +1,125 @@
+# `wasmcloud-wasi-kv`
+
+This repository contains a [WebAssembly System Interface (WASI) key-value contract][wasi-kv] compliant Web Assembly module which can be composed into a "macro" or "composite" component to provide key value functionality.
+
+This module is special because it implements the KV contract for any and all [wasmCloud][wasmcloud] environments, showcasing the reusability of WASI, WIT ,  it's .
+
+## ❓ What does that even mean?
+
+- [WebAssembly ("WASM")][wasm] brings amazing sandboxing and almost-C/C++/Rust speeds
+- Compiling your favorite `$LANGUAGE` to WebAssembly modules means you can run anywhere WASM modules run
+- WASI introduces a framework for extending the functionality available to any given WebAssembly sandbox
+- [Wasm Interface Type ("WIT")][wit] is the language (an [IDL][idl], like gRPC/Thrift/etc) for describing functionality
+- The [WASI key-value contract][wasi-kv] describes operations a cache key value store (think [Redis][redis]) would offer
+- If a WebAssembly module implements the key-value contract, it can be *composed*/*combined* with another module that *depends on* that contract.
```

Comment:
I'm going to suggest some rewordings here because I'm having a hard time following, and I think there's a lot of work to do for someone first seeing these notions. I'd love to see these bullets be something like (and please feel free to wordsmith)

- WebAssembly / Wasm is a compilation target for many languages that creates a platform agnostic binary that can run on any CPU architecture, operating system, or even in the browser
- Webassembly System interface (WASI) is a framework that extends platform-specific APIs to the WebAssembly sandbox, allowing Wasm to do things like operate on a filesystem or make use of network sockets
- [Wasm Interface Type ("WIT")][wit] is an interface definition language for describing functionality that WebAssembly modules can use
- The WASI key-value contract describes basic operations (get, set, del) for interacting with a key-value store. This can be implemented in many different ways, this example uses Redis

TL;DR: WebAssembly components allows you to use a common interface to write an application that uses a key-value store: in a language of your choice, with the implementation of your choice, and run it on any platform

Reply:
Really looking to @ricochet and @autodidaddict here too, this is an important step as people first hit this repository and need to understand components / wasmcloud / wit

Reply:
Just refactored! let me know what you think @brooksmtownsend

Reply:
I really like the new bullets. Great stuff @vados-cosmonic